### ISSUE TITLE
Improve handling of file associations, setting Sumatra as default PDF reader

### ISSUE THREAD

**Issue by kjk** [sumatrapdfreader/sumatrapdf#1248]
This is a master bug for improving how Sumatra handles file associations for formats it supports.
Things to do, before 3.2 or shortly after:

- [ ] audit which extension we claim to support, make sure that we include everything that we support
- [ ] for supported extensions, installer will write necessary register entries so that Sumatra shows up in "Open With" explorer menu for those extensions
- [] on Win 10 we need to check on Sumatra startup if it's registered as default app for PDF. If not, ask the user. If the user says yes, launch the settings panel. Remember if user said "no".
- [] the same will happen when trigerred from settings dialog

Maybe:
- [] allow setting Sumatra as default reader for other files (like EPUB, CBZ etc.) The exact UI fro this is TBD

**Comment by GitHubRulesOK** [comment]
@kjk I know you have been reluctant to extend the Icon previewer beyond PDF however the times I have seen the extra formats active it has appeared stable and been of benefit. You will get inevitable complaints/issues like the ones saying there is no zoom etc. but having the ability to preview image files (even including old ones like TGA) and e-book contents is a great boon.

**Comment by kpym** [comment]
It will be nice if not only Sumatra could be chosen as default app to open all supported file types, but also if Sumatra could be chosen (independently) as default app for previewing this files in the preview pane of the explorer. 

This is already the case for PDF but having it for EPUB, MOBI, DJVU, ... would be great.

And, IMO, it is important that this features :
- are independent : so that we could chose sumatra as default app for some extensions and as default previewer for others,
- could be set/unset without uninstall/install : accessible through the options,
- restorable : Sumatra should save the previous default app in a way to be able to restore it later.

**Comment by GitHubRulesOK** [comment]
Setting SumatraPDF as default for PDF has been a bug bear but does not stop the need to independantly associate it with different supported file types.

The previewer file type association is a seperate issue as Windows 10 will now preview images (.png .tif etc. and I think that includes .tga) it also allows previewing .(o)xps in a zoomable fashion.

Where SumatraPDF previewer has the edge :-) is that it can scroll through multi-page Gif and Tif images and can preview many many more filetypes.

Currently there are two related open issues one for extending CBR support to v5 https://github.com/sumatrapdfreader/sumatrapdf/issues/1676 , the other is previewing uncompressed FB2 files https://github.com/sumatrapdfreader/sumatrapdf/issues/1677

An additional requested feature is to add (e)PS preview support https://github.com/sumatrapdfreader/sumatrapdf/issues/840

Here is a snapshot of how extensive the supported filetypes can be
![image](https://user-images.githubusercontent.com/12754913/87369254-f88d2080-c577-11ea-8fe8-277e1b55028e.png)

**Comment by GitHubRulesOK** [comment]
@kjk 
core issue of associations is now different on win 10/11 requiring hashes and seeing you have removed non viable code, perhaps review the OP requirement?

**Comment by kjk** [comment]
Win10+ doesn't allow for programmatic setting of default apps.

We do the best we can: register SumatraPDF with shell as supporting file extensions we know about.

That makes SumatraPDF listed in `Open With` explorer menu and and option when double-clicking and allows user to set SumatraPDF as default handler for those extensions.

**Comment by solarjoe** [comment]
I was not able to make SumatraPDF the default viewer with all the methods mentioned (Open with, Default App, etc.).
No matter what, the pdf still opened with Adobe. 

The solution I came across was to set Adobe to be default viewer. Yes, actually. I went to the Adobe settings and clicked "Use as default PDF viewer" and ticked "Notify if Adobe is not the default viewer". No matter why, but after this Sumatra was set as default. Something was really off.
